containers/utils/detect_image.py

```diff
@@ -0,0 +1,147 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Example using TF Lite to detect objects in a given image."""
+
+import argparse
+import time
+
+from PIL import Image
+from PIL import ImageDraw
+import cv2
+
+import detect
+import tflite_runtime.interpreter as tflite
+import platform
+from cscore import CameraServer, VideoSource, UsbCamera, MjpegServer
+from networktables import NetworkTablesInstance
+import numpy as np
+
+# TODO Make me right
+team = 1519
+
+
+def load_labels(path, encoding='utf-8'):
+    """Loads labels from file (with or without index numbers).
+
+    Args:
+      path: path to label file.
+      encoding: label file encoding.
+    Returns:
+      Dictionary mapping indices to labels.
+    """
+    with open(path, 'r', encoding=encoding) as f:
+        lines = f.readlines()
+        if not lines:
+            return {}
+
+        if lines[0].split(' ', maxsplit=1)[0].isdigit():
+            pairs = [line.split(' ', maxsplit=1) for line in lines]
+            return {int(index): label.strip() for index, label in pairs}
+        else:
+            return {index: line.strip() for index, line in enumerate(lines)}
+
+
+def make_interpreter(model_file):
+    model_file, *device = model_file.split('@')
+    return tflite.Interpreter(
+        model_path=model_file,
+        experimental_delegates=[
+            tflite.load_delegate("libedgetpu.so.1",
+                                 {'device': device[0]} if device else {})
+        ])
+
+
+def draw_objects(draw, objs, labels):
+    """Draws the bounding box and label for each object."""
+    for obj in objs:
+        bbox = obj.bbox
+        draw.rectangle([(bbox.xmin, bbox.ymin), (bbox.xmax, bbox.ymax)],
+                       outline='red')
+        draw.text((bbox.xmin + 10, bbox.ymin + 10),
+                  '%s\n%.2f' % (labels.get(obj.id, obj.id), obj.score),
+                  fill='red')
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-m', '--model', required=True,
+                        help='File path of .tflite file.')
+    parser.add_argument('-i', '--input', required=True,
+                        help='File path of image to process.')
+    parser.add_argument('-l', '--labels',
+                        help='File path of labels file.')
+    parser.add_argument('-t', '--threshold', type=float, default=0.4,
+                        help='Score threshold for detected objects.')
+    parser.add_argument('-o', '--output',
+                        help='File path for the result image with annotations')
+    parser.add_argument('-n', '--team', type=int,
+                        help='Team number')
+    args = parser.parse_args()
+
+    # Init cameras
+    WIDTH, HEIGHT = 720, 1920
```

Comment:
Lil sus. If you look here: https://github.com/wpilibsuite/WPILibPi/releases/download/v2021.2.1/python-multiCameraServer.zip you can see how to automatically get camera resolution